File: js/pyramid.js
const pyramidBuilder = n => {
  const midpoint = Math.floor((2 * n - 1) / 2);

  for (let row = 0; row < n; row++) {
    let level = "";
    for (let column = 0; column < 2 * n - 1; column++) {
      level += midpoint - row <= column && midpoint + row >= column ? "#" : " ";
    }
    console.log(level);
  }
};

const pyramidBuilderRecursive = (
  n,
  row = 0,
  column = 0,
  level = "",
  midpoint = Math.floor((2 * n - 1) / 2)
) => {
  if (row === n) {
    return;
  }

  if (column === 2 * n - 1) {
    console.log(level);
    return pyramidBuilderRecursive(n, row + 1, 0, "", midpoint);
  }

  level += midpoint - row <= column && midpoint + row >= column ? "#" : " ";
  return pyramidBuilderRecursive(n, row, column + 1, level, midpoint);
};

pyramidBuilderRecursive(4);

File: python/pyramid.py
import math


def pyramidBuilder(n):
    midpoint = math.floor((2 * n - 1) / 2)
    for row in range(n):
        level = ''
        for column in range(2 * n - 1):
            level += '#' if midpoint - row <= column and midpoint + row >= column else ' '
        print(level)


def pyramidBuilderRecursive(n, row=0, column=0, level=''):
    midpoint = math.floor((2 * n - 1) / 2)
    if row == n:
        return
    if column == 2 * n - 1:
        print(level)
        return pyramidBuilderRecursive(n, row + 1, 0, '')
    level += '#' if midpoint - row <= column and midpoint + row >= column else ' '
    return pyramidBuilderRecursive(n, row, column + 1, level)


pyramidBuilderRecursive(4)
